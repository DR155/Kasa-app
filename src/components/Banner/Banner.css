.banner-container {
  position: relative;
  width: 100%;
  height: 223px;
  border-radius: 25px;
  overflow: hidden;
  margin-bottom: 43px;
}

.banner-image {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
}

/* Overlay et texte uniquement pour la bannière avec texte (ex: Home) */
.banner-container:not(.banner-no-text) .banner-overlay {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.3); /* Assombrissement pour lisibilité */
  border-radius: 25px; /* Doit correspondre à .banner-container */
}

.banner-text {
  color: white;
  font-size: 48px;
  font-weight: 500;
  text-align: center;
  padding: 0 20px; /* Évite que le texte ne touche les bords */
  line-height: 1.2;
}


/* --- Media Queries --- */

@media (max-width: 767.98px) {
  .banner-container {
    height: 111px;
    border-radius: 10px;
    margin-bottom: 22px;
  }

  .banner-container:not(.banner-no-text) .banner-overlay {
    border-radius: 10px;
    justify-content: flex-start;
  }

  .banner-text {
    font-size: 24px;
    text-align: left;
    padding: 0 16px;
  }
}