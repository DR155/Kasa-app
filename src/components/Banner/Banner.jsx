import React from 'react';
import './Banner.css';

function Banner({ imageSrc, text, altText = "Banni√®re", showTextOverlay = false }) {
  const bannerClassName = `banner-container ${!showTextOverlay ? 'banner-no-text' : ''}`;

  return (
    <div className={bannerClassName}>
      <img src={imageSrc} alt={altText} className="banner-image" />
      {showTextOverlay && text && (
        <div className="banner-overlay">
          <h1 className="banner-text">{text}</h1>
        </div>
      )}
    </div>
  );
}

export default Banner;