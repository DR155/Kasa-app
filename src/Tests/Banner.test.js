import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Banner from '../components/Banner/Banner.jsx'; 
import homeBannerImage from '../assets/images/homeBanner.jpg';

// Mock de l'image pour les tests
const mockImageSrc = "test-banner.jpg";
const mockAltText = "Bannière de test";
const mockText = "Texte de la bannière";

describe('Composant Banner', () => {
  test('doit rendre l\'image avec le bon src et alt', () => {
    render(<Banner imageSrc={mockImageSrc} altText={mockAltText} />);
    const imageElement = screen.getByRole('img');
    expect(imageElement).toHaveAttribute('src', mockImageSrc);
    expect(imageElement).toHaveAttribute('alt', mockAltText);
  });

  test('doit afficher le texte et l\'overlay si showTextOverlay est true et text est fourni', () => {
    render(
      <Banner
        imageSrc={mockImageSrc}
        altText={mockAltText}
        text={mockText}
        showTextOverlay={true}
      />
    );
    expect(screen.getByText(mockText)).toBeInTheDocument();
    // Vérifie la présence de l'overlay (tu peux ajouter une data-testid à l'overlay si besoin)
    const overlay = screen.getByText(mockText).closest('.banner-overlay'); // Ou une autre façon de cibler l'overlay
    expect(overlay).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: mockText })).toBeInTheDocument(); // Le texte est un h1
  });

  test('ne doit pas afficher le texte ni l\'overlay si showTextOverlay est false', () => {
    render(
      <Banner
        imageSrc={mockImageSrc}
        altText={mockAltText}
        text={mockText}
        showTextOverlay={false}
      />
    );
    expect(screen.queryByText(mockText)).not.toBeInTheDocument();
    // Pour l'overlay, si sa présence dépend du texte, cette assertion suffit.
    // Sinon, il faudrait une manière plus directe de le cibler.
    const bannerContainer = screen.getByRole('img').closest('.banner-container');
    expect(bannerContainer).toHaveClass('banner-no-text'); // Vérifie la classe conditionnelle
  });

  test('ne doit pas afficher le texte si text n\'est pas fourni, même si showTextOverlay est true', () => {
    render(
      <Banner
        imageSrc={mockImageSrc}
        altText={mockAltText}
        showTextOverlay={true}
        // text prop est omise
      />
    );
    expect(screen.queryByRole('heading')).not.toBeInTheDocument(); // Aucun h1 ne devrait être présent
  });

  test('doit correspondre au snapshot avec texte', () => {
    const { container } = render(
      <Banner
        imageSrc={mockImageSrc}
        altText={mockAltText}
        text={mockText}
        showTextOverlay={true}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('doit correspondre au snapshot sans texte', () => {
    const { container } = render(
      <Banner
        imageSrc={mockImageSrc}
        altText={mockAltText}
        showTextOverlay={false}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});