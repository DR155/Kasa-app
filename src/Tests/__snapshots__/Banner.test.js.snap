// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Composant Banner doit correspondre au snapshot avec texte 1`] = `
<div
  class="banner-container "
>
  <img
    alt="Bannière de test"
    class="banner-image"
    src="test-banner.jpg"
  />
  <div
    class="banner-overlay"
  >
    <h1
      class="banner-text"
    >
      Texte de la bannière
    </h1>
  </div>
</div>
`;

exports[`Composant Banner doit correspondre au snapshot sans texte 1`] = `
<div
  class="banner-container banner-no-text"
>
  <img
    alt="Bannière de test"
    class="banner-image"
    src="test-banner.jpg"
  />
</div>
`;
